src/pages/shipping.js

```diff
@@ -407,83 +352,68 @@ export const shippingManifest = async (boxIdArray, userName, isTempMonitorInclud
-                <button type="button" class="btn btn-primary" data-dismiss="modal" id="printBox">Print Box Manifest</button>
-            </div>
-            <div style="float:left;width: 33%;" id="boxManifestCol3">
-                <button type="button" class="btn btn-primary" data-dismiss="modal" id="completePackaging">Packaging Complete</button>
-            </div>
-        </div>
-        `;*/
-    removeActiveClass('navbar-btn', 'active')
+
+    removeActiveClass('navbar-btn', 'active');
     const navBarBtn = document.getElementById('navBarReviewShipmentContents');
     navBarBtn.classList.add('active');
     document.getElementById('contentBody').innerHTML = template;
 
-    if(isTempMonitorIncluded){
-        populateTempSelect(boxIdArray);
-    }
-
-    document.getElementById('shippingHiddenTable').innerHTML = JSON.stringify(boxIdAndBagsObj);
-    
-    
-    //document.getElementById('boxManifestTable').appendChild(result);
-    
-    populateShippingManifestHeader(boxIdAndBagsObjToDisplay, userName, location, site, currShippingLocationNumber); // populate shipping header via site specfiic location selected from shipping page
+    populateShippingManifestHeader(userName, siteAcronym, currShippingLocationNumber);
     populateShippingManifestBody(boxIdAndBagsObjToDisplay);
     addEventNavBarShipment("navBarShippingDash", userName);
-    await populateTempCheck();
-    const btn = document.getElementById('assignTrackingNumberPage'); // assignTracking
-    addEventShipPrintManifest('printBox')
+    const btn = document.getElementById('assignTrackingNumberPage');
+    addEventShipPrintManifest('printBox');
     addEventNavBarShipment('returnToPackaging', userName);
 
-    
-
-    document.getElementById('assignTrackingNumberPage').addEventListener('click', e => {
+    btn && btn.addEventListener('click', async (e) => {
         e.stopPropagation();
-        if(btn.classList.contains('active')) return;
-
         const tempBoxElement = document.getElementById('tempBox');
-
         if (isTempMonitorIncluded && tempBoxElement.value === '') {
-            showNotifications({title: 'Missing field!', body: 'Please enter the box where the temperature monitor is being stored.'}, true)
+            showNotifications({title: 'Missing field!', body: 'Please enter the box where the temperature monitor is being stored.'}, true);
             return;
         }
 
         let boxWithTempMonitor = '';
-        if( isTempMonitorIncluded){
+        if (isTempMonitorIncluded) {
             boxWithTempMonitor = tempBoxElement.value;
         }
 
-        shipmentTracking(boxIdAndBagsObjToDisplay, userName, boxWithTempMonitor);
+        await shipmentTracking(boxIdAndBagsObjToDisplay, userName, boxWithTempMonitor);
     });
-    //addEventNavBarShipment("navBarShippingDash");
-    //addEventBackToSearch('backToSearch');
-}
 
+    hideAnimation();
+
+    function tempSelectStringRender({boxIdArray, isTempMonitorIncluded}) {
```

Comment:
This is a question on my end. Is the reasoning to have the `tempSelectStringRender` function declaration inside the `shippingManifest` function because the `tempSelectStringRender` function is called within and we assume that the `tempSelectStringRender` function will be used once?

Reply:
It required me search around to find out that, although used in `shippingManifest`, `tempBox` is defined in another file. I wanted to bring the related UI definitions closer and make things more straightforward. Putting the newly defined function in `shippingManifest` is one option.

But defining `tempSelectStringRender` inside `shippingManifest` makes `shippingManifest` unnecessarily bulky and less pretty. So I'm now moving it out.

Likely `tempSelectStringRender` will be called more than once during a shipping.

Reply:
Thanks for the response and changing a perspective of mine when dealing with function declarations.

> It required me search around to find out that, although used in `shippingManifest`, `tempBox` is defined in another file. I wanted to bring the related UI definitions closer and make things more straightforward. Putting the newly defined function in `shippingManifest` is one option.

I normally move nested functions outside of the function expressions out of habit. But after seeing this approach, it changed my perspective that a function declaration nested inside a function can work. I think I would keep `tempSelectStringRender` inside `shippingManifest` if it was less bulky and was only used within `shippingManifest`, the scenario of grouping UI definitions.

> Likely `tempSelectStringRender` will be called more than once during a shipment.

Sorry I think I misspoke, I meant the `tempSelectStringRender` function being referenced in other locations in the code outside of the `shippingManifest` function.